src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -24,38 +26,91 @@ internal class ProcessMetrics
+
         // TODO: change to ObservableUpDownCounter
         MeterInstance.CreateObservableGauge(
             "process.memory.usage",
-            () =>
-            {
-                CurrentProcess.Refresh();
-                return CurrentProcess.WorkingSet64;
-            },
+            () => values.GetMemoryUsage(),
             unit: "By",
             description: "The amount of physical memory in use.");
 
         // TODO: change to ObservableUpDownCounter
         MeterInstance.CreateObservableGauge(
             "process.memory.virtual",
-            () =>
-            {
-                CurrentProcess.Refresh();
-                return CurrentProcess.VirtualMemorySize64;
-            },
+            () => values.GetVirtualMemoryUsage(),
             unit: "By",
             description: "The amount of committed virtual memory.");
+
+        if (options.CpuStatesEnabled == false)
+        {
+            MeterInstance.CreateObservableCounter(
+            $"process.cpu.time",
+            () => values.GetProcessorCpuTime(),
+            unit: "s",
+            description: "Total CPU seconds.");
+        }
+        else
+        {
+            MeterInstance.CreateObservableCounter(
+            $"process.cpu.time",
+            () => values.GetProcessorCpuTimeWithBreakdown(),
+            unit: "s",
+            description: "Total CPU seconds broken down by different states.");
+        }
     }
 
-    /// <summary>
-    /// Initializes a new instance of the <see cref="ProcessMetrics"/> class.
-    /// </summary>
-    /// <param name="options">The options to define the metrics.</param>
-    public ProcessMetrics(ProcessInstrumentationOptions options)
+    private class InstrumentsValues
     {
+        private readonly Diagnostics.Process currentProcess = Diagnostics.Process.GetCurrentProcess();
+
+        public InstrumentsValues()
+        {
+            this.currentProcess.Refresh();
+        }
+
+        private enum CpuState
+        {
+            System,
+            User,
+            Wait,
+        }
+
+        public long GetMemoryUsage()
+        {
+            return this.currentProcess.WorkingSet64;
+        }
+
+        public long GetVirtualMemoryUsage()
+        {
+            return this.currentProcess.VirtualMemorySize64;
+        }
+
+        public long GetProcessorCpuTime()
+        {
+            return this.currentProcess.TotalProcessorTime.Seconds;
+        }
+
+        // See spec:
+        // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/process-metrics.md#metric-instruments
+        public Measurement<long>[] GetProcessorCpuTimeWithBreakdown()
+        {
+            Measurement<long>[] measurements = new Measurement<long>[Enum.GetNames(typeof(CpuState)).Length];
+            var priviledgedCpuTime = this.currentProcess.PrivilegedProcessorTime.Seconds;
+            var userCpuTime = this.currentProcess.UserProcessorTime.Seconds;
```

Comment:
Do we want integer or double?

Reply:
Updated to use int.

Reply:
Why int?

Reply:
well, because the `.Seconds` from the Timespan struct returns int?

Reply:
> well, because the `.Seconds` from the Timespan struct returns int?

How is this related?

Reply:
Shouldn't it be `.TotalSeconds`?